src/Components/animation/DisplayedCuts.js

```diff
@@ -21,83 +21,58 @@ class DisplayedCuts extends React.Component {
 
   _displayCut = cut => {
     return (
-      <View key={cut.key + 4000} style={[StyleSheet.absoluteFill]} height="100%" width="100%">
+      <View key={cut.key}>
         <Animated.View
           style={[
-            { height: 2 },
-            { width: cut.d1 },
-            { borderRadius: 1 },
-            { borderWidth: 1 },
-            { borderColor: 'green' },
-            { borderStyle: 'dashed' },
-            { position: 'absolute' },
-            { top: cut.top1 },
-            { left: cut.left1 },
-            { transform: [{ rotate: cut.angle1.__getValue() }] },
+            styles.cut,
+            {
+              width: cut.d1,
+              top: cut.top1,
+              left: cut.left1,
+              transform: [{ rotate: cut.angle1.__getValue() }],
+            },
           ]}
         />
-
         <Animated.View
           style={[
-            { height: 1 },
-            { width: cut.d2 },
-            { borderRadius: 1 },
-            { borderWidth: 1 },
-            { borderColor: 'green' },
-            { borderStyle: 'dashed' },
-            { position: 'absolute' },
-            { top: cut.top2 },
-            { left: cut.left2 },
-            { transform: [{ rotate: cut.angle2.__getValue() }] },
+            styles.cut,
+            {
+              width: cut.d2,
+              top: cut.top2,
+              left: cut.left2,
+              transform: [{ rotate: cut.angle2.__getValue() }],
+            },
           ]}
         />
-        {cut.cutCircle !== null && cut.cutCircle !== undefined ? cut.cutCircle.render() : undefined}
-        {cut.countercutCircle !== null && cut.countercutCircle !== undefined
-          ? cut.countercutCircle.render()
-          : undefined}
+        {cut?.cutCircle}
+        {cut?.countercutCircle}
       </View>
     );
   };
 
   render() {
     return (
-      <Animated.View key="1" style={[{ position: 'absolute', left: 0, top: 0 }]} height="100%" width="100%">
-        {this.state.stateFromProps.cuts !== undefined && this.state.stateFromProps.cuts !== null
-          ? this.state.stateFromProps.cuts[this.state.step]?.map(item => {
-              return (
-                <Animated.View
-                  key={item.key}
-                  style={[{ position: 'absolute', left: 0, top: 0 }]}
-                  height="100%"
-                  width="100%"
-                >
-                  {this.state.stateFromProps.cuts[this.state.step]?.map(this._displayCut)}
-                </Animated.View>
-              );
-            })
-          : undefined}
+      <Animated.View style={{ position: 'absolute', left: 0, top: 0 }} height="100%" width="100%">
+        {this.state.stateFromProps?.cuts[this.state.step]?.map(this._displayCut)}
       </Animated.View>
     );
   }
 
   /** Used to update the cuts when a modification is made in the animation */
   static getDerivedStateFromProps(props, state) {
     // Test if the animation has changed
-    var isEqual = true;
```

Comment:
I used const and let instead of var because it makes it clear which ones are not updated (const) and the scope of let is simpler to get while reading the code